torch/distributed/_composable/fsdp/_fsdp_state.py

```diff
@@ -235,7 +244,13 @@ def _register_pre_backward_hook(self, output: Any) -> Any:
             t for t in flat_outputs if (torch.is_tensor(t) and t.requires_grad)
         )
         if tensors:
-            register_multi_grad_hook(tensors, self._pre_backward, mode="any")
+            # Invariant: if Compiled Autograd is used, regardless of whether forward is eager mode,
+            # we must use the Traceable FSDP codepath.
+            if not torch._dynamo.compiled_autograd.compiled_autograd_enabled:
```

Comment:
Is this flag going to be True even if users only wrap the `loss.sum().backward()` with CompiledAutograd?

Reply:
For Traceable FSDP case, from my experiment it seems we need to wrap the forward with CompiledAutograd as well.

I feel that it might also be a good convention to use going forward (i.e. "if you want to use PT2-D, please wrap both forward and backward with CompiledAutograd").